lib/GenomeFileUtil/core/GenomeInterface.py

```diff
@@ -141,13 +148,95 @@ def save_one_genome(self, params):
+        solr_core = 'taxonomy_ci'
+        query = '/select?q=scientific_name:"{}"&fl=scientific_name%2Cscientific_lineage%2Ctaxonomy_id%2Cdomain&rows=5&wt=json'
+        match = re.match("\S+\s?\S*", scientific_name)
+        if not match:
+            return default
+        res = requests.get(solr_url+solr_core+query.format(match.group(0)))
+        results = res.json()['response']['docs']
+        if not results:
+            return default
+        taxonomy = results[0]['scientific_lineage']
+        taxon_reference = '{}/{}_taxon'.format(
+            taxon_wsname, results[0]['taxonomy_id'])
+        domain = results[0]['domain']
+
+        return taxonomy, taxon_reference, domain
+
+    @staticmethod
+    def determine_tier(source):
+        """
+        Given a user provided source parameter, assign a source and genome tier
+        """
+        low_source = source.lower()
+        if 'refseq' in low_source:
+            if 'Reference' in low_source:
+                return "Refseq", ['Reference', 'Representative',
+                                  'ExternalDB']
+            if 'Representative' in low_source:
+                return "Refseq", ['Representative', 'ExternalDB']
+            return "Refseq", ['ExternalDB']
+        if 'phytozome' in low_source:
+            if 'flagship' in source:
+                return "Phytosome", ['Reference', 'Representative',
+                                     'ExternalDB']
+            return "Phytosome", ['Representative', 'ExternalDB']
+        if 'ensembl' in low_source:
+            return "Ensembl", ['Representative', 'ExternalDB']
+        return source, ['User']
+
+    @staticmethod
+    def validate_genome(genome, print_size=True):
+        """
+        Run a series of checks on the genome object and return any warnings
+        """
+        def _get_size(obj):
+            return sys.getsizeof(json.dumps(obj))
+        allowed_tiers = {'Representative', 'Reference', 'ExternalDB', 'User'}
+
+        log('Validating genome object contents')
+        warnings = []
+        if len(genome.get('cdss', [])) < len(genome['features']):
+            warnings.append("CDS array should be at at least as long as the "
+                            "Features array.")
+        if genome['domain'] == "Bacteria" and len(genome.get('cdss', [])) \
+                != len(genome['features']):
+            warnings.append("For prokaryotes, CDS array should be the same "
+                            "length as the Features array.")
+        if len(genome.get('mrnas', [])) != len(genome.get('cdss', [])):
+            warnings.append("mRNA array should be the same length as the CDS"
+                            "array if present.")
+        if "molecule_type" in genome and genome['molecule_type'] != "DNA":
+            warnings.append("Genome molecule_type is expected to be DNA but is"
+                            "actually " + str(genome['molecule_type']))
```

Comment:
If molecule type is "DNA" or "ds-DNA" everything is fine. If the molecule type is "ss-RNA","ds-RNA" or "ss-DNA", then we need to confirm that the domain is either Virus or Viroid. If it is not we need to do a warning/error. If it is none of those we need to do a warning/error.  The behavior will probably be determined by running in strict or loose mode.

Reply:
Also more checks in the PD, but they can be added in a later iteration.